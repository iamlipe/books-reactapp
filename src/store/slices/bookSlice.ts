import {createSlice, PayloadAction} from '@reduxjs/toolkit';
import {AxiosError} from 'axios';

export interface Book {
  id: string;
  authors: string[];
  title: string;
  imageUrl: string;
  pageCount: number;
  publisher: string;
  published: number;
}

export interface BookDetailsResponse {
  authors: string[];
  title: string;
  description: string;
  pageCount: number;
  category: string;
  imageUrl: string;
  language: string;
  isbn10: string;
  isbn13: string;
  publisher: string;
  published: number;
  id: string;
}

export interface BooksResponse {
  data: Book[];
  page: number;
  totalItems: number;
  totalPages: number;
}

export interface BooksRequest {
  page?: string;
  amount?: string;
  title?: string;
  category?: string;
}

export interface BookDetailsRequest {
  id: string;
}

interface BookState {
  isLoading: boolean;
  error: AxiosError | null;
  statusCode: number | null;

  allBooks: BooksResponse | null;
  detailsBook: BookDetailsResponse | null;
}

const initialState: BookState = {
  isLoading: false,
  error: null,
  statusCode: null,

  allBooks: null,
  detailsBook: null,
};

const bookSlice = createSlice({
  name: 'book',
  initialState,
  reducers: {
    GET_BOOKS: (state, _: PayloadAction<BooksRequest>) => ({
      ...state,
      isLoading: true,
      error: null,
      statusCode: null,
    }),
    GET_BOOKS_SUCCESS: (
      state,
      {
        payload: {data, status},
      }: PayloadAction<{data: BooksResponse; status: number}>,
    ) => ({
      ...state,
      isLoading: false,
      error: null,
      statusCode: status,

      allBooks: data,
    }),
    GET_BOOKS_FAILURE: (state, {payload: {error}}) => ({
      ...state,
      isLoading: false,
      error,
    }),

    GET_DETAILS_BOOK: (state, _: PayloadAction<BookDetailsRequest>) => ({
      ...state,
      isLoading: true,
      error: null,
      statusCode: null,
    }),

    GET_DETAILS_BOOK_SUCCESS: (
      state,
      {
        payload: {data, status},
      }: PayloadAction<{data: BookDetailsResponse; status: number}>,
    ) => ({
      ...state,
      isLoading: false,
      error: null,
      statusCode: status,

      detailsBook: data,
    }),

    GET_DETAILS_BOOK_FAILURE: (state, {payload: {error}}) => ({
      ...state,
      isLoading: false,
      error,
    }),

    REMOVE_DETAILS_BOOK: state => ({
      ...state,

      detailsBook: null,
    }),
  },
});

const {actions, reducer} = bookSlice;

export const bookState = initialState;

export const {
  GET_BOOKS,
  GET_BOOKS_SUCCESS,
  GET_BOOKS_FAILURE,
  GET_DETAILS_BOOK,
  GET_DETAILS_BOOK_SUCCESS,
  GET_DETAILS_BOOK_FAILURE,
  REMOVE_DETAILS_BOOK,
} = actions;

export default reducer;
