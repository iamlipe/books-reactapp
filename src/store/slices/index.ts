import {combineReducers} from '@reduxjs/toolkit';

import user, {userState} from './userSlice';
import book, {bookState} from './bookSlice';

export const globalState = {user: userState, book: bookState};

export const globalReducer = {user, book};

const rootReducer = combineReducers(globalReducer);

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
