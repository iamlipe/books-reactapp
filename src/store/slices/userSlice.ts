import {createSlice, PayloadAction} from '@reduxjs/toolkit';
import {AxiosError} from 'axios';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  id: string;
  name: string;
  birthdate: string;
  gender: string;
  email: string;
}

export interface User {
  id: string;
  name: string;
  email: string;
  authorization: string | undefined;
  refreshToken: string | undefined;
}

interface UserState {
  isLoading: boolean;
  auth: User | null;
  error: AxiosError | null;
  statusCode: number | null;
}

const initialState: UserState = {
  isLoading: false,
  auth: null,
  error: null,
  statusCode: null,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    LOGIN: (state, _: PayloadAction<LoginRequest>) => ({
      ...state,
      isLoading: true,
      error: null,
      statusCode: null,
    }),
    LOGIN_SUCCESS: (
      state,
      {payload: {data, status}}: PayloadAction<{data: User; status: number}>,
    ) => ({
      ...state,
      isLoading: false,
      auth: data,
      statusCode: status,
    }),
    LOGIN_FAILURE: (state, {payload: {error}}) => ({
      ...state,
      isLoading: false,
      error,
    }),

    LOGOUT: (state, _: PayloadAction) => ({
      ...state,
      isLoading: true,
      error: null,
    }),
    LOGOUT_SUCCESS: () => ({
      ...initialState,
    }),
    LOGOUT_FAILURE: (state, {payload: {error}}) => ({
      ...state,
      isLoading: false,
      error,
    }),
  },
});

const {actions, reducer} = userSlice;

export const userState = initialState;

export const {
  LOGIN,
  LOGIN_FAILURE,
  LOGIN_SUCCESS,
  LOGOUT,
  LOGOUT_SUCCESS,
  LOGOUT_FAILURE,
} = actions;
export default reducer;
