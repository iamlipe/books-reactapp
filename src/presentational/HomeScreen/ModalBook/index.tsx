/* eslint-disable react-native/no-inline-styles */
import React, {memo} from 'react';
import styled from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {Dimensions} from 'react-native';
import {Overlay} from 'react-native-elements';
import {RFValue} from 'react-native-responsive-fontsize';
import {REMOVE_DETAILS_BOOK} from '@store/slices/bookSlice';

import {useReduxSelector} from '@hooks/useReduxSelector';
import {useReduxDispatch} from '@hooks/useReduxDispatch';

import Aspas from '@assets/svgs/aspas.svg';

const {width, height} = Dimensions.get('window');

const ModalBook = () => {
  const {detailsBook} = useReduxSelector(state => state.book);
  const dispatch = useReduxDispatch();

  function handleRemoveDetailsBook() {
    dispatch(REMOVE_DETAILS_BOOK());
  }

  return (
    <>
      {detailsBook && <StyledBackground />}
      <StyledOverlay
        isVisible={!!detailsBook}
        onBackdropPress={handleRemoveDetailsBook}
        backdropStyle={{backgroundColor: 'transparent'}}
        fullScreen={false}
        overlayStyle={{
          width: width * 0.9,
          height: height * 0.85,
          marginTop: height * 0.05,
        }}>
        <StyledBaseButton onPress={handleRemoveDetailsBook}>
          <Icon name="close" size={16} color="#333" />
        </StyledBaseButton>

        <StyledContainerScroll showsVerticalScrollIndicator={false}>
          <StyledImage source={{uri: detailsBook?.imageUrl}} />

          <StyledTextTitle>
            {detailsBook && detailsBook.title.length > 35
              ? `${detailsBook?.title.substring(0, 35)}...`
              : detailsBook?.title}
          </StyledTextTitle>

          <StyledRowAuthors>
            {detailsBook?.authors.map((author, index) => (
              <StyledTextAuthors key={index}>
                {detailsBook.authors.length - 1 !== index
                  ? `${author}, `
                  : author}
              </StyledTextAuthors>
            ))}
          </StyledRowAuthors>

          <StyledTextSectionTitle>Informações</StyledTextSectionTitle>

          <StyledRowInfo>
            <StyledTextInfoTitle>Páginas</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.pageCount}</StyledTextInfo>
          </StyledRowInfo>

          <StyledRowInfo>
            <StyledTextInfoTitle>Editora</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.publisher}</StyledTextInfo>
          </StyledRowInfo>

          <StyledRowInfo>
            <StyledTextInfoTitle>Publicação</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.published}</StyledTextInfo>
          </StyledRowInfo>

          <StyledRowInfo>
            <StyledTextInfoTitle>Idioma</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.language}</StyledTextInfo>
          </StyledRowInfo>

          <StyledRowInfo>
            <StyledTextInfoTitle>Título Original</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.title}</StyledTextInfo>
          </StyledRowInfo>

          <StyledRowInfo>
            <StyledTextInfoTitle>ISBN-10</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.isbn10}</StyledTextInfo>
          </StyledRowInfo>

          <StyledRowInfo>
            <StyledTextInfoTitle>ISBN-13</StyledTextInfoTitle>
            <StyledTextInfo>{detailsBook?.isbn13}</StyledTextInfo>
          </StyledRowInfo>

          <StyledTextSectionTitle>Resenha da editora</StyledTextSectionTitle>
          <StyledRowDescription>
            <StyledTextInfo>
              <StyledAspas /> {`  ${detailsBook?.description}`}
            </StyledTextInfo>
          </StyledRowDescription>
        </StyledContainerScroll>
      </StyledOverlay>
    </>
  );
};

const StyledBackground = styled.View`
  position: absolute;
  left: -20px;

  z-index: 1;

  width: ${width}px;
  height: ${height}px;

  background-color: ${({theme}) => theme.colors.BLACK};

  opacity: 0.4;
`;

const StyledOverlay = styled(Overlay)``;

const StyledImage = styled.Image`
  height: 412px;
  width: 100%;

  align-self: center;

  margin-bottom: 24px;
`;

const StyledContainerScroll = styled.ScrollView`
  height: 200px;

  padding: 16px;
`;

const StyledRowInfo = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const StyledRowAuthors = styled(StyledRowInfo)`
  justify-content: flex-start;
`;

const StyledRowDescription = styled(StyledRowAuthors)``;

const StyledAspas = styled(Aspas)``;

const StyledTextTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.HEEBO_MEDIUM};
  font-size: ${RFValue(28)}px;

  color: ${({theme}) => theme.colors.TEXT_DARK};
`;

const StyledTextAuthors = styled.Text`
  font-family: ${({theme}) => theme.fonts.HEEBO_REGULAR};
  font-size: ${RFValue(12)}px;

  color: ${({theme}) => theme.colors.TEXT_PRIMARY};
`;

const StyledTextSectionTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.HEEBO_MEDIUM};
  font-size: ${RFValue(12)}px;

  color: ${({theme}) => theme.colors.TEXT_DARK};

  text-transform: uppercase;

  margin: 32px 0 12px 0;
`;

const StyledTextInfoTitle = styled(StyledTextSectionTitle)`
  line-height: 20px;

  text-transform: none;

  margin: 0;
`;

const StyledTextInfo = styled.Text`
  font-family: ${({theme}) => theme.fonts.HEEBO_REGULAR};
  font-size: ${RFValue(12)}px;

  color: ${({theme}) => theme.colors.TEXT_GRAY};
`;

const StyledBaseButton = styled.TouchableOpacity`
  position: absolute;
  top: ${-height * 0.06}px;
  right: 0;

  width: 32px;
  height: 32px;

  align-items: center;
  justify-content: center;

  background-color: ${({theme}) => theme.colors.WHITE};

  border-radius: 16px;
  border: 1px solid ${({theme}) => theme.colors.BLACK_TRANSPARENT};
`;

export default memo(ModalBook);
