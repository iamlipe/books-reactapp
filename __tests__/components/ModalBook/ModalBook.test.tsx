import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@config/functions/Testing';
import {BookDetailsResponse} from '@store/slices/bookSlice';

import ModalBook from '@presentational/HomeScreen/ModalBook';

describe('ModalBook', () => {
  const detailsBook: BookDetailsResponse = {
    id: '1',
    title: 'Memórias Póstumas de Brás Cubas',
    authors: ['Machado de Assis'],
    imageUrl:
      'https://images-na.ssl-images-amazon.com/images/I/41ZS6DFo+vL._SX328_BO1,204,203,200_.jpg',
    pageCount: 159,
    publisher: 'Saraiva',
    published: 2019,
    category: 'romance',
    description:
      'A publicação desta obra não só inaugura o Realismo no Brasil, como inicia a etapa mais complexa da obra de Machado de Assis. Com ela, aprofunda-se a sua análise da realidade e refina-se a sua linguagem, sendo considerada a obra que prenuncia algumas técnicas da literatura moderna.',
    language: 'português',
    isbn10: '8502082191',
    isbn13: '978-8502082199',
  };

  test('should render correctly', () => {
    renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
  });

  test('should match snapshot', () => {
    matchSnapshotWithProvider(<ModalBook />, {
      book: {detailsBook},
    });
  });

  test('should render imageBook', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });

    const imageBook = modalBook.queryByTestId('image-book-modal');

    expect(imageBook).toBeTruthy();
  });

  test('should render title', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });

    const titleModalBook = modalBook.queryAllByText(detailsBook.title);

    expect(titleModalBook.length).toEqual(2);
  });

  test('should render authors', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });

    detailsBook.authors.forEach(author => {
      const authorModalBook = modalBook.queryByText(author);

      expect(authorModalBook).toBeTruthy();
    });
  });

  test('should render Paperback', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const paperbackModalBook = modalBook.queryByText(
      `${detailsBook.pageCount}`,
    );

    expect(paperbackModalBook).toBeTruthy();
  });

  test('should render Publisher', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const publisherModalBook = modalBook.queryByText(detailsBook.publisher);

    expect(publisherModalBook).toBeTruthy();
  });

  test('should render Published', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const publishedModalBook = modalBook.queryByText(
      `${detailsBook.published}`,
    );

    expect(publishedModalBook).toBeTruthy();
  });

  test('should render Language', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const languageModalBook = modalBook.queryByText(detailsBook.language);

    expect(languageModalBook).toBeTruthy();
  });

  test('should render isbn10', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const isbn10ModalBook = modalBook.queryByText(detailsBook.isbn10);

    expect(isbn10ModalBook).toBeTruthy();
  });

  test('should render isbn13', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const isbn13ModalBook = modalBook.queryByText(detailsBook.isbn13);

    expect(isbn13ModalBook).toBeTruthy();
  });

  test('should render Description', () => {
    const modalBook = renderWithThemeProvider(<ModalBook />, {
      book: {detailsBook},
    });
    const descriptionModalBook = modalBook.queryByText(detailsBook.description);

    expect(descriptionModalBook).toBeTruthy();
  });
});
