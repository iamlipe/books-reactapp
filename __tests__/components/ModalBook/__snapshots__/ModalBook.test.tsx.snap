// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBook should match snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#333333",
          "height": 1334,
          "left": -20,
          "opacity": 0.4,
          "position": "absolute",
          "width": 750,
          "zIndex": 1,
        },
      ]
    }
  />,
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Array [
        Object {},
      ]
    }
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="RNE__Overlay__backdrop"
    />
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 3,
            "height": 1133.8999999999999,
            "marginTop": 66.7,
            "padding": 10,
            "shadowColor": "rgba(0, 0, 0, .3)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowRadius": 4,
            "width": 675,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#33333333",
              "borderRadius": 16,
              "borderStyle": "solid",
              "borderWidth": 1,
              "height": 32,
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": -80.03999999999999,
              "width": 32,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#333",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <RCTScrollView
          showsVerticalScrollIndicator={false}
          style={
            Array [
              Object {
                "height": 200,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
            ]
          }
        >
          <View>
            <Image
              source={
                Object {
                  "uri": "https://images-na.ssl-images-amazon.com/images/I/41ZS6DFo+vL._SX328_BO1,204,203,200_.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "height": 412,
                    "marginBottom": 24,
                    "width": "100%",
                  },
                ]
              }
              testID="image-book-modal"
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 55,
                  },
                ]
              }
            >
              Memórias Póstumas de Brás Cubas
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#AB2680",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                Machado de Assis
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 24,
                    "marginBottom": 12,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 32,
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              Informações
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                Páginas
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                159
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                Editora
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                Saraiva
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                Publicação
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                2019
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                Idioma
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                português
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                Título Original
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                Memórias Póstumas de Brás Cubas
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                ISBN-10
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                8502082191
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontFamily": "Heebo-Medium",
                      "fontSize": 24,
                      "lineHeight": 20,
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "marginTop": 0,
                      "textTransform": "none",
                    },
                  ]
                }
              >
                ISBN-13
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                978-8502082199
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 24,
                    "marginBottom": 12,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 32,
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              Resenha da editora
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#999999",
                      "fontFamily": "Heebo-Regular",
                      "fontSize": 24,
                    },
                  ]
                }
              >
                <SvgMock
                  style={
                    Array [
                      Object {},
                    ]
                  }
                />
                 
                  A publicação desta obra não só inaugura o Realismo no Brasil, como inicia a etapa mais complexa da obra de Machado de Assis. Com ela, aprofunda-se a sua análise da realidade e refina-se a sua linguagem, sendo considerada a obra que prenuncia algumas técnicas da literatura moderna.
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </Modal>,
]
`;
